src/cargo/sources/registry/mod.rs

```diff
@@ -497,9 +497,17 @@ impl<'cfg> RegistrySource<'cfg> {
             }
 
             // Once that's verified, unpack the entry as usual.
-            entry
-                .unpack_in(parent)
-                .chain_err(|| format!("failed to unpack entry at `{}`", entry_path.display()))?;
+            entry.unpack_in(parent).chain_err(|| {
+                if cfg!(windows) {
+                    if restricted_names::is_windows_reserved_path(entry_path.as_path()) {
+                        format!(
```

Comment:
This doesn't have any effect, since `format!` just evaluates to a string, it doesn't return from the closure.